### ISSUE TITLE
Feature Request: Reverse/Undo "Mark as spam and delete".

### ISSUE THREAD

**Issue by chris001** [webmin/usermin#81]
It can happen, that a user accidentally clicks "Mark as spam and delete" on genuine real mail, because it's such a common reflex for mailboxes that receive a lot of spam mail.
So, it would be nice to have a feature to undo that accidental "Mark as spam and delete" action.  Because you don't want all future email from that sender to be (incorrectly) classified as spam because one time you mistakenly clicked on the wrong action, "Mark as spam and delete", when you should have just clicked on "Delete" instead.
Would this be possible without much effort or too complicated?

**Comment by chris001** [comment]
Ah OK.  Just for consideration, possibly will help at some point.  Not so much un-deleting the message, but un-mark the email From address as a "spam sender".  Something like, go into the page where anti-spam engines (spam assassin, razor, pyzor, etc), show the history of all/recent messages that were "marked as spam and deleted", obviously the message is deleted and doesn't exist anymore, however the metadata should be memorized by the anti-spam engines in some way (email address From, etc).  A way to select the metadata of the recent item marked as spam, and click "unmark as spam", so that the anti-spam engine will treat it as a normal message sender, and not automatically redirect it into a spam/junk folder.  Does this "undo mark as spam" feature, or something equivalent, already exist?

**Comment by ksihota** [comment]
I know this is over a year old but this is how I see the workflow should go.
1. User receives a message they feel is Spam
2. The message is moved to a Learn Spam folder
3. Using an improved Folder Auto Clearing tool all messages in the folder are marked and Deleted

![261440572-fb451013-3599-4240-992a-1abd91825dbf](https://github.com/webmin/usermin/assets/29150649/833e39b3-84e2-4c67-b0da-eceb4231695c)

This would allow a user to change their mind before the task was completed since the auto clearing would be programmed to occur at their preferred time/date etc.
It would eliminate the accidental, unrecoverable marking and deleting of messages.

This also allows users who are using a different email client to perform the Mark as Spam and Delete function. All they would need to do in Usermin would be to define the folder and the Auto Clearing rule

**Comment by jcameron** [comment]
Yeah, that's not a bad idea ... I'll think about it. Also I don't want to go off in a totally different direction to more common mail apps like Gmail..

**Comment by ksihota** [comment]
Just went to my Gmail account to see how it compares to Thunderbird and Usermin. I have never used the Gmail web client.

WOW, I hate it. Way too complicated for users unless they are willing to spend a lot of time trying to figure it out. Although, I have to admit that it is extremely responsive/quick in displaying data.

It uses labels for organizing messages which could be a good idea as messages can fit into multiple groups, but it certainly complicates things for those who want a simple system. I expect that only a few people really need to keep years of messages organized for quick retrieval.
A year or so ago I had thought I had cleaned up my Gmail account, removing useless messages and organizing the ones I wanted to keep. When I checked the messages in GMail I found that none of them were gone. There are over 33,000 messages, most of which are garbage using over 3GB of storage space.
Since I hadn't used the Gmail client but instead relied on Thunderbird IMAP I guess what was happening was that instead of deleting messages all that was happening was labels were being applied and then removed. All Mail still retained the original messages. I'm still trying to wrap my head around the process but I think I have figured it out and am going to reattempt to clean the account using my Thunderbird client. 

My guess is that most older users would think of a mail message as a single entity that could only exists in one place and would conceptualize the storage of these single entities in different folders. Younger users who have grown up using tags etc or users more comfortable with relational database systems would conceptualize mail in term of organization rather than single, physical entities. Even though I am familiar with both systems I have always viewed mail messages in the old model of physical mail. It is a much simpler way to relate to the messages but does not provide the same flexibility as using labels or tags. Personally I believe most users want simplicity in their mail/messages. I also believe that, really, its only those who manage massive amounts of data for multiple purposes that prefer and require the more complex/capable/flexible system of tags and labels.

I believe that Google should have done a better job when it implemented its labelling system in order to make it more friendly to the original mail model. They certainly need to do a way better job simplifying their interface. Or maybe its Thunderbird's problem because of how they set up their own mailboxes when creating a GMail account, that does not deal with the Gmail system of labels properly. In my mind those issues need to be handled by the developers and not left to users to try and figure out.

Not sure how Usermin handles this.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.

**Comment by ksihota** [comment]
From what I can see there has been no update to the Automatic Folder Clearing. It is an option that I would like but not sure if its anything others would use. The automatic Folder Clearing option is a nice feature to help automatically keep your mailbox organized. The additional features would just supplement what already exists.
